src/core/src/core_logic/Stopwatch.py

```diff
@@ -63,11 +65,24 @@ def stop(self):
         # Rounding off to one digit after decimal e.g. 14.574372666666667 will become 14.6
         self.time_taken_in_secs = round(self.time_taken_in_secs, 1)
 
+    # Stop the stopwatch, set end_time and write details in telemetry. Create new end_time even if end_time is already set
     def stop_and_write_telemetry(self, message):
         self.stop()
         self.set_task_details(message)
         self.composite_logger.log("Stopwatch details: " + self.task_details)
 
+    # Write stopwatch details in telemetry. Use the existing end_time if it is already set otherwise set new end_time
+    def write_telemetry_for_stopwatch(self, message):
+        if self.end_time is None:
+            self.composite_logger.log_debug(str(Stopwatch.StopwatchException.NOT_STOPPED))
+            self.end_time = self.env_layer.datetime.datetime_utcnow()
+        if self.start_time is None:
+            self.composite_logger.log_debug(str(Stopwatch.StopwatchException.NOT_STARTED))
+            self.start_time = self.end_time
+        self.time_taken = self.env_layer.datetime.total_minutes_from_time_delta(self.end_time - self.start_time)
```

Comment:
It should be self.time_taken_in_secs. Also need to call function  total_seconds_from_time_delta and round off to 1 digit after decimal
This variable was renamed in another commit so need to update here.

Reply:
Done.